package-res/resources/web/test/prompting/PromptPanelSpec.js

```diff
@@ -1363,6 +1363,24 @@ define([ 'dojo/number', 'dojo/i18n', 'common-ui/prompting/PromptPanel',
           expect(paramDefn.page).toBe(1);
         });
       });
+
+      describe("setParamDefn", function() {
+        it("should set a parameter definition", function() {
+          var onStateChangedSpy = jasmine.createSpy("onStateChanged");
+          panel.onStateChanged = onStateChangedSpy;
+          panel.paramDefn = null;
+
+          panel.setParamDefn(paramDefn);
+
+          expect(panel.paramDefn).toBe(paramDefn);
+          expect(panel.onStateChanged).toHaveBeenCalledWith("promptNeeded", undefined, paramDefn.promptNeeded);
+          expect(panel.onStateChanged).toHaveBeenCalledWith("paginate", undefined, paramDefn.paginate);
+          expect(panel.onStateChanged).toHaveBeenCalledWith("totalPages", undefined, paramDefn.totalPages);
+          expect(panel.onStateChanged).toHaveBeenCalledWith("showParameterUI", undefined,paramDefn.showParameterUI());
+          expect(panel.onStateChanged).toHaveBeenCalledWith("allowAutoSubmit", undefined, paramDefn.allowAutoSubmit());
+          expect(panel.onStateChanged).toHaveBeenCalledWith("page", undefined, paramDefn.page);
+        });
+      });
```

Comment:
In addition better add one more test to check that onStateChanged callback is not called.